pystac/__init__.py

```diff
@@ -3,59 +3,72 @@
 """
 
 # flake8: noqa
-
-
-class STACError(Exception):
-    """A STACError is raised for errors relating to STAC, e.g. for
-    invalid formats or trying to operate on a STAC that does not have
-    the required information available.
-    """
-    pass
-
-
-from pystac.version import (__version__, get_stac_version, set_stac_version)
-from pystac.stac_io import STAC_IO
-from pystac.extensions import Extensions
-from pystac.stac_object import (STACObject, STACObjectType)
-from pystac.media_type import MediaType
-from pystac.link import (Link, HIERARCHICAL_LINKS)
-from pystac.catalog import (Catalog, CatalogType)
-from pystac.collection import (Collection, Extent, SpatialExtent, TemporalExtent, Provider)
-from pystac.item import (Item, Asset, CommonMetadata)
-
-from pystac.serialization import stac_object_from_dict
+from pystac.errors import (
+    STACError,  # type:ignore
```

Comment:
Needs a comment explaining why type annotations should be ignored.

Reply:
FYI, this throws a type error due to unused imports

Reply:
Ok, that makes sense.  Hopefully the tooling catches up to this as it looks to be a common complaints.

- https://stackoverflow.com/questions/31079047/python-pep8-class-in-init-imported-but-not-used
- https://github.com/SublimeLinter/SublimeLinter/issues/167
- https://github.com/PyCQA/pyflakes/issues/162

I'm not sure how I feel about adding things to `__all__` as suggested here in [PEP 8 Public and Internal Interfaces](https://www.python.org/dev/peps/pep-0008/#public-and-internal-interfaces)

> To better support introspection, modules should explicitly declare the names in their public API using the __all__ attribute. Setting __all__ to an empty list indicates that the module has no public API.